src/types/hashed.ts

```diff
@@ -21,14 +20,13 @@ export const types = {
 	},
 };
 
-export const validate = TypeUtils.validate.checkRequired(value => {
+export const validate = TypeUtils.validate.checkRequired(async (value) => {
 	if (typeof value !== 'string') {
 		throw new Error('is not a string');
 	} else {
-		return Promise.resolve(bcrypt.genSalt()).then(salt =>
-			bcrypt.hash(value, salt),
-		);
+		const salt = await bcrypt.genSalt();
+		return bcrypt.hash(value, salt);
 	}
 });
 
-export const compare = Promise.method(bcrypt.compare.bind(bcrypt));
+export const compare = bcrypt.compare.bind(bcrypt);
```

Comment:
Is `bcrypt.compare` an async function? Otherwise it changes the type of the exported `compare`. Just thought I'd mention in case this slipped.

Reply:
Yeah it returns a native promise which is why I wrapped it for back-compat previously